cfpb/cfpb-chart-builder
# New line comparison chart

Welcome to the world of Mortgage Performance.

We need a chart that can accept an arbitrary number of data sources and render them on the same line chart. I created a new chart type called "line-comparison" for this purpose.

I also started ES6'ifying things. I didn't convert everything -- I just wrote new code in ES6. cfgov-refresh (and all new CFPB projects) are ES6-friendly.

Also: I created a new canary branch for development purposes. Because I'll be adding lots of scary new features I'll open MP pull requests against canary and we can periodically merge canary into master.

Current /test/index.html page: https://cfpb.github.io/cfpb-chart-builder/
New /test/index.html page: https://codepen.io/contolini/full/OjgmKo/ *Same as the previous one but with two line comparison charts at the top. The first chart will use the new `update()` method to change its data sources after five seconds.*

## Additions

- New `line-comparison` chart. It accepts a semicolon-separated listed of sources and will render each one as a line on the chart.
- Created and exposed a `createChart()` method so that charts can be programmatically created instead of only on document ready (see the README).
- Created and exposed an `update()` method that Chart instances can use to update themselves.

## Removals

-

## Changes

- Pulled the XHR functionality into its own file `get-data.js` and wrote some tests for it. It's now Promise-based.
- Updated the README with config options.
- Updated gulp tasks to pass JS through Babel.

## Testing

- Pull down this branch.
- `nvm use 8`
- `./setup.sh` to build everything
- `npm test` to verify unit and browser tests pass as normal (you'll need to add [sauce credentials](https://github.com/cfpb/cfpb-chart-builder#testing) if you haven't already
- `gulp watch` and look at the two new line comparison charts at the top of the page.

To verify the charts still load correctly in cfgov-refresh:

- Pull down this branch.
- `nvm use 8`
- `npm link`
- `cd ~/wherever/your/projects/are/cfgov-refresh/`
- `npm link cfpb-chart-builder`
- `gulp build`
- `./runserver`
- Verify that [this page](http://127.0.0.1:8000/data-research/consumer-credit-trends/credit-cards/borrower-risk-profiles/) and [this page](https://www.consumerfinance.gov/data-research/consumer-credit-trends/credit-cards/borrower-risk-profiles/) are identical and that the charts aren't borked.

## Screenshots

![screen shot 2017-08-10 at 1 51 24 pm](https://user-images.githubusercontent.com/1060248/29184079-0cf0425e-7dd3-11e7-97b9-8bd1b8cc49c5.png)

## Notes

-

## Todos

- ~~I'll soon be adding an `updateChart()` method to redraw charts.~~ This has been added to this PR.

## Checklist
